hail/src/main/scala/is/hail/io/plink/ExportPlink.scala

```diff
@@ -104,15 +104,17 @@ object ExportPlink {
         }
       }
 
-      Iterator.single(rowCount)
-    }.collect().sum
+      Iterator.single(f -> rowCount)
+    }.collect().foldLeft(
+      (FastIndexedSeq[String](), 0L)
+    ) { case ((files, sum), (f, n)) => (files :+ f, sum + n) }
```

Comment:
This is potentially quadratic

Reply:
Is `FastIndexedSeq` really that dumb?

Reply:
```
  def :+[B >: A, That](elem: B)(implicit bf: CanBuildFrom[Repr, B, That]): That = {
    val b = bf(repr)
    b ++= thisCollection
    b += elem
    b.result()
  }
```

Reply:
oh dear jesus the implementation of WrappedArray.canBuildFrom makes me want to cry.

Reply:
![image](https://user-images.githubusercontent.com/10562794/49764722-3e984900-fc9e-11e8-937a-69e9f1154110.png)